packages/elements-core/src/components/Docs/HttpOperation/HttpOperation.tsx

```diff
@@ -34,28 +36,28 @@ const HttpOperationComponent = React.memo<HttpOperationProps>(
     const prettyName = (data.summary || data.iid || '').trim();
     const hasBadges = isDeprecated || isInternal;
 
-    const header = (!layoutOptions?.noHeading || hasBadges) && (
-      <VStack spacing={5}>
-        <HStack spacing={5}>
-          {!layoutOptions?.noHeading && prettyName ? (
-            <Heading size={1} fontWeight="semibold">
-              {prettyName}
-            </Heading>
-          ) : null}
-
-          <HStack spacing={2}>
-            {isDeprecated && <DeprecatedBadge />}
-            {isInternal && <InternalBadge isHttpService />}
-          </HStack>
-        </HStack>
-
-        <MethodPath method={data.method} path={data.path} />
-      </VStack>
+    const header = (
+      <OperationHeader
+        id={data.id}
+        method={data.method}
+        path={data.path}
+        noHeading={layoutOptions?.noHeading}
+        hasBadges={hasBadges}
+        name={prettyName}
+        isDeprecated={isDeprecated}
+        isInternal={isInternal}
+      />
     );
 
+    const descriptionChanged = nodeHasChanged?.({ nodeId: data.id, attr: 'description' });
     const description = (
       <VStack spacing={10}>
-        {data.description && <MarkdownViewer className="HttpOperation__Description" markdown={data.description} />}
+        {data.description && (
+          <Box pos="relative">
+            <MarkdownViewer className="HttpOperation__Description" markdown={data.description} />
+            <NodeAnnotation change={descriptionChanged} />
```

Comment:
Ditto on these `NodeAnnotation` components - they just return `null` if `change` is undefined, which it will be for all existing usage - so it does nothing - these changes overall are quite low risk / unobtrusive.